pkg/services/syncer/syncer.go

```diff
@@ -90,6 +94,7 @@ func syncNamespace(config config.Config, namespace string) error {
 
 		vers, err := s.Sync(config, cc)
 		if err != nil {
+			config.Recorder.Event(&cc, corev1.EventTypeWarning, "sync", err.Error())
```

Comment:
so linking with the comment above, why in here we are not passing a copy but a reference? will the Event modify the original object then?

Reply:
no, the difference from the above case is that we were modifying the status before updating it, that's why we took the DeepCopy, it should be safe here

Reply:
to wrap up (a generic rule of thumb) you deepcopy when you are about to change something that needs to be posted back at Kube: 
```
copy := moc.DeepCopy()
copy.Status.Status = "..."
//// then do something with copy, as posting an update and such
```

of course there might be other cases too, but this is generally a good way to go. You see usually also deepcopy intentionally also at beginning to avoid forgetting it by mistake

Reply:
ok, got it, thanks for the explanation!